manager/role_manager.go

```diff
@@ -62,13 +82,29 @@ func (rm *roleManager) Run(ctx context.Context) {
 	if err != nil {
 		log.G(ctx).WithError(err).Error("failed to check nodes for role changes")
 	} else {
+		// Put all members in the removed list first - if the node object exists
```

Comment:
unrelated: Please also add a high level comment on what all of this method is trying to achieve.